### PR TITLE
Adds new T3 Gorger

### PR DESCRIPTION
**Pull request by Walarks**
## About The Pull Request

Adds new T3 Gorger

This xeno is supposed to achieve the niche of being a tank that can take marine fire for extended periods of time, while providing a lot of healing for other xenos and itself. To do so it needs to attack marines to gain and store blood, which forces it to expose itself to gain the full effect of its abilities.

I am also open to giving these skills in one form or another to other xenos, but I've personally not thought of any that they fit on.

TO DO:
Xeno sprite
Ability icons

Stats:
Health - 600 - 800
Resistances - 5 - 20%
Speed - same as defiler
Plasma - 100 - 150
Slash Damage - 15 - 20

Abilities:
 - Blood Bank - stores up to 100 blood that is used to enable some of its abilities
 - Drain - 50 plasma, 15s CD - immobilise a marine for 2s and causes it to drop held items. After 2s deals 20 damage (not mitigated by armor) and gain 10 blood. If the channel is broken, the marine also gets freed earlier
 - Rejuvenate/Transfusion:
If cast on another xeno - 5 blood, 2s CD - restores 30% health
If cast on a dead human - no cost, 2s CD - restores health, plasma and sunder at a rate as if resting on weeds. If standing on weeds, this goes on top of the passive restoration benefits.
If cast on itself - 10 blood, 20s CD - over 5s restores 40% health and full plasma
 - Carnage - 100 plasma, 30s CD - attacking heals for damage dealt and gives 5 blood
 - Feast - available at 100 blood, 3m CD - drains 5 blood and restores 10% health per second. Carnage also heals all other xenos in 4 tiles for 70% of damage dealt

## Why It's Good For The Game

I think this xeno's playstyle as a tank that heals/makes tankier xenos around them in exchange of having to be in-combat isn't currently found among other xenos. Defender is the closest that I can think of, but to me it seems to be more of a T1 warrior with an extra quirk than a tank. Crusher is also tanky but the skills/stats don't fit with its current kit.

## Changelog
:cl:Slotbot
add: Adds new xeno caste
/:cl:

### CODE REVIEW

**Comment by SplinterGP** [comment]
Carnage slash has 2 linter errors, check them at checks.

**Comment by Chaznoodles** [comment]
Did you mean Gorger, or Gouger? This ain't it chief

**Review by TiviPlus — CHANGES_REQUESTED** [review]
Needs to either be rav additions or  get a better sprite that isn't a defiler recolor

**Comment by spookydonut** [comment]
Will people stop trying to add new castes instead of replacing/improving existing castes?

**Comment by Walarks** [comment]
> Will people stop trying to add new castes instead of replacing/improving existing castes?

I am ok with replacing Ravager with it as Ravager is a big Hunter with more damage/stuns and without stealth.

**Comment by Walarks** [comment]
Drafting until sprite

**Comment by BeloneX** [comment]
George.

File: code/modules/mob/living/carbon/xenomorph/castes/gorger/abilities_gorger.dm [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+	if(!.)
+		return
+	if(!A.can_sting())
+		to_chat(owner, "<span class='warning'>This won't do!</span>")
+		return FALSE
+	if(X.blood_bank >= 100)
+		to_chat(owner, "<span class='warning'>Somehow we feel sated for now...</span>")
+		return FALSE
+	if(get_dist(owner, A) > 1)
+		to_chat(owner, "<span class='warning'>It is outside of our reach! We need to be</span>")
```
Comment (XSlayer300) [inline code/modules/mob/living/carbon/xenomorph/castes/gorger/abilities_gorger.dm:24]:
```suggestion
		to_chat(owner, "<span class='warning'>It is outside of our reach! We need to be closer!</span>")
```

File: code/modules/mob/living/carbon/xenomorph/castes/gorger/abilities_gorger.dm [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+		if(get_dist(X, target) == -1)
+			if(!COOLDOWN_CHECK(src, rejuvenate_self_cooldown))
+				to_chat(X, "<span class='notice'>We need another [round(COOLDOWN_TIMELEFT(src, rejuvenate_self_cooldown) / 10)] seconds before we can revitalize ourselves.</span>")
+				return FALSE
+			if(X.blood_bank < 10)
+				to_chat(X, "<span class='notice'>We need [10 - X.blood_bank]u more blood to revitalize ourselves.</span>")
+				return FALSE
+		else
+			if(!X.line_of_sight(target) || get_dist(X, target) > 2)
+				to_chat(X, "<span class='notice'>Beyond our reach, we must be close and our way clear.</span>")
```
Comment (XSlayer300) [inline code/modules/mob/living/carbon/xenomorph/castes/gorger/abilities_gorger.dm:88]:
```suggestion
				to_chat(X, "<span class='notice'>It is beyond our reach, we must be close and our way clear.</span>")
```

File: code/modules/mob/living/carbon/xenomorph/castes/gorger/castedatum_gorger.dm [hunk 3]
```diff
@@ -0,0 +1,130 @@
+/datum/xeno_caste/gorger
+	caste_name = "Gorger"
+	display_name = "Gorger"
+	upgrade_name = ""
+	caste_desc = "A frightening looking, bulky xeno that drips with a familiar red fluid."
```
Comment (XSlayer300) [inline code/modules/mob/living/carbon/xenomorph/castes/gorger/castedatum_gorger.dm:5]:
```suggestion
	caste_desc = "A frightening looking, bulky alien creature that drips with a familiar red fluid."
```

File: code/modules/mob/living/carbon/xenomorph/castes/gorger/castedatum_gorger.dm [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+	)
+
+/datum/xeno_caste/gorger/young
+	upgrade_name = "Young"
+
+	upgrade = XENO_UPGRADE_ZERO
+
+/datum/xeno_caste/gorger/mature
+	upgrade_name = "Mature"
+	caste_desc = "A frightening looking, bulky xeno that drips with a familiar red fluid. It looks a little more dangerous."
```
Comment (XSlayer300) [inline code/modules/mob/living/carbon/xenomorph/castes/gorger/castedatum_gorger.dm:53]:
```suggestion
	caste_desc = "A frightening looking, bulky alien creature that drips with a familiar red fluid. It looks a little more dangerous."
```

File: code/modules/mob/living/carbon/xenomorph/castes/gorger/castedatum_gorger.dm [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+
+	// *** Evolution *** //
+	upgrade_threshold = 500
+
+	// *** Defense *** //
+	soft_armor = list("melee" = 10, "bullet" = 10, "laser" = 10, "energy" = 10, "bomb" = XENO_BOMB_RESIST_0, "bio" = 10, "rad" = 10, "fire" = 10, "acid" = 10)
+
+/datum/xeno_caste/gorger/elder
+	upgrade_name = "Elder"
+	caste_desc = "A frightening looking, bulky xeno that drips with a familiar red fluid. It looks pretty strong."
```
Comment (XSlayer300) [inline code/modules/mob/living/carbon/xenomorph/castes/gorger/castedatum_gorger.dm:78]:
```suggestion
	caste_desc = "A frightening looking, bulky alien creature that drips with a familiar red fluid. It looks pretty strong."
```

**Comment by Walarks** [comment]
reeeeeeeeeeeeeee

**Comment by Al-1ce** [comment]
Thought this was pretty cool, Tivi was being a bit of a shitter about the sprite anyways.

**Pull request closed without merging**